streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -193,6 +195,22 @@ public int hashCode() {
     public TopologyBuilder() {}
 
     /**
+     * Set the applicationId to be used for auto-generated internal topics.
+     *
+     * This is required before calling {@link #sourceTopics}, {@link #topicGroups},
+     * {@link #copartitionSources} and {@link #build(Integer)}.
+     *
+     * @param applicationId the streams applicationId. Should be the same as set by
+     * {@link org.apache.kafka.streams.StreamsConfig#APPLICATION_ID_CONFIG}
+     */
+    public synchronized final TopologyBuilder setApplicationId(String applicationId) {
```

Comment:
I'm making the same change as in trunk in another PR: https://github.com/apache/kafka/pull/1736 for API consistency.